### DISCUSSION TITLE
python3.9[21666]: KeyError: <StickerFormatType.unknown_4: 4>

### DISCUSSION THREAD

**Discussion by ornelasf1** [Rapptz/discord.py#9703] [Q&A] [answered]
Im seeing the following stacktrace on my discord.py bot for some reason. It seems to be starting at bot.run() so that leads me to believe it's coming from the library. Anyone having a similar issue?

> Jan 07 18:58:36 googlefeud python3.9[21666]: Traceback (most recent call last):
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/discord_google_feud/main.py", line 322, in <module>
Jan 07 18:58:36 googlefeud python3.9[21666]:     bot.run(TOKEN)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/client.py", line 828, in run
Jan 07 18:58:36 googlefeud python3.9[21666]:     asyncio.run(runner())
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
Jan 07 18:58:36 googlefeud python3.9[21666]:     return loop.run_until_complete(main)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
Jan 07 18:58:36 googlefeud python3.9[21666]:     return future.result()
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/client.py", line 817, in runner
Jan 07 18:58:36 googlefeud python3.9[21666]:     await self.start(token, reconnect=reconnect)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/client.py", line 746, in start
Jan 07 18:58:36 googlefeud python3.9[21666]:     await self.connect(reconnect=reconnect)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/client.py", line 627, in connect
Jan 07 18:58:36 googlefeud python3.9[21666]:     await self.ws.poll_event()
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/gateway.py", line 622, in poll_event
Jan 07 18:58:36 googlefeud python3.9[21666]:     await self.received_message(msg.data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/gateway.py", line 572, in received_message
Jan 07 18:58:36 googlefeud python3.9[21666]:     func(data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/state.py", line 1209, in parse_guild_create
Jan 07 18:58:36 googlefeud python3.9[21666]:     guild = self._get_create_guild(data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/state.py", line 1147, in _get_create_guild
Jan 07 18:58:36 googlefeud python3.9[21666]:     guild._from_data(data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/guild.py", line 487, in _from_data
Jan 07 18:58:36 googlefeud python3.9[21666]:     self.stickers: Tuple[GuildSticker, ...] = tuple(
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/guild.py", line 488, in <lambda>
Jan 07 18:58:36 googlefeud python3.9[21666]:     map(lambda d: state.store_sticker(self, d), guild.get('stickers', []))
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/state.py", line 379, in store_sticker
Jan 07 18:58:36 googlefeud python3.9[21666]:     self._stickers[sticker_id] = sticker = GuildSticker(state=self, data=data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/sticker.py", line 273, in __init__
Jan 07 18:58:36 googlefeud python3.9[21666]:     self._from_data(data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/sticker.py", line 416, in _from_data
Jan 07 18:58:36 googlefeud python3.9[21666]:     super()._from_data(data)
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/sticker.py", line 280, in _from_data
Jan 07 18:58:36 googlefeud python3.9[21666]:     self.url: str = f'{Asset.BASE}/stickers/{self.id}.{self.format.file_extension}'
Jan 07 18:58:36 googlefeud python3.9[21666]:   File "/home/gfeudadmin/.local/lib/python3.9/site-packages/discord/enums.py", line 534, in file_extension
Jan 07 18:58:36 googlefeud python3.9[21666]:     return lookup[self]
Jan 07 18:58:36 googlefeud python3.9[21666]: KeyError: <StickerFormatType.unknown_4: 4>

**Comment by Rapptz** [comment, marked as answer]
Your library is almost a year out of date. Please update the library before making issues or discussions.

**Reply by ornelasf1** [reply]
Right, my bad. It's working now
